manifests/init.pp

```diff
@@ -26,148 +26,148 @@
-  $daemon_group                                              = undef,
-  $global_owner                                              = $::nginx::params::global_owner,
-  $global_group                                              = $::nginx::params::global_group,
-  $global_mode                                               = $::nginx::params::global_mode,
-  $log_dir                                                   = $::nginx::params::log_dir,
-  $log_group                                                 = $::nginx::params::log_group,
-  $log_mode                                                  = '0750',
-  Variant[String, Array[String]] $http_access_log            = "${log_dir}/${::nginx::params::http_access_log_file}",
-  $http_format_log                                           = undef,
-  Variant[String, Array[String]] $nginx_error_log            = "${log_dir}/${::nginx::params::nginx_error_log_file}",
-  Nginx::ErrorLogSeverity $nginx_error_log_severity          = 'error',
-  $pid                                                       = $::nginx::params::pid,
-  $proxy_temp_path                                           = $::nginx::params::proxy_temp_path,
-  $root_group                                                = $::nginx::params::root_group,
-  $run_dir                                                   = $::nginx::params::run_dir,
-  $sites_available_owner                                     = $::nginx::params::sites_available_owner,
-  $sites_available_group                                     = $::nginx::params::sites_available_group,
-  $sites_available_mode                                      = $::nginx::params::sites_available_mode,
-  Boolean $super_user                                        = $::nginx::params::super_user,
-  $temp_dir                                                  = $::nginx::params::temp_dir,
-  Boolean $server_purge                                      = false,
+  String[1] $client_body_temp_path                                              = $::nginx::params::client_body_temp_path,
+  Boolean $confd_only                                                           = false,
+  Boolean $confd_purge                                                          = false,
+  String[1] $conf_dir                                                           = $::nginx::params::conf_dir,
+  Optional[Nginx::Toggle] $daemon                                               = undef,
+  String[1] $daemon_user                                                        = $::nginx::params::daemon_user,
+  Optional[String[1]] $daemon_group                                             = undef,
+  String[1] $global_owner                                                       = $::nginx::params::global_owner,
+  String[1] $global_group                                                       = $::nginx::params::global_group,
+  Stdlib::Filemode $global_mode                                                 = $::nginx::params::global_mode,
```

Comment:
great that we finally have a datatype for this

Reply:
Can you, in a seperate PR, bump the stdlib requirement to at least 4.22.0 in metadata.json?

![2018-04-21-205849_608x197_scrot](https://user-images.githubusercontent.com/341866/39087821-e4dacfc2-45a6-11e8-96b4-f4ff6b6ff970.png)